### PR TITLE
Rs/wx/normalize space schedules

### PR DESCRIPTION
**Pull request by weilixu**

### CODE REVIEW

**Comment by charliepnnl** [comment]
I notice that the RDS claims that this function is designed to "determine a normalized schedule for a data element in space".  But its definition seems to heavily depend on the details of the `interior_lighting` data element.  Should this function be called `normalize_interior_lighting_schedules`?

File: rct229/ruleset_functions/normalize_space_schedules.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            space_total_hourly_use_per_area_array = [0.0] * len(
+                hourly_use_per_area_array
+            )
+
+        # make sure the hourly schedules have the same number of hours
+        assert_(
+            len(space_total_hourly_use_per_area_array)
+            == len(hourly_use_per_area_array),
+            f"Number of hours mismatched. The space has number of hours of: {len(space_total_hourly_use_per_area_array)} "
+            f"but the schedule id {interior_lighting['lighting_multiplier_schedule']} has number of hours of: "
```
Comment (charliepnnl) [inline rct229/ruleset_functions/normalize_space_schedules.py:76]: Is this proper Python syntax to concatenate these strings?  I think you need to put a `+` between them.

**Review by charliepnnl — APPROVED** [review]
Approved with comments that should be addressed.

**Review by yunjoonjung-PNNL — APPROVED** [review]

File: rct229/data_fns/table_G3_7_fns.py [hunk 2]
```diff
@@ -113,17 +114,27 @@
...
 }
 
+FULL_AUTO_ON = schema_enums["LightingOccupancyControlType"].FULL_AUTO_ON.name
+PARTIAL_AUTO_ON = schema_enums["LightingOccupancyControlType"].PARTIAL_AUTO_ON.name
+MANUAL_ON = schema_enums["LightingOccupancyControlType"].MANUAL_ON.name
+
 # ATRIUM_LOW_MEDIUM
-def table_G3_7_lookup(lighting_space_type, space_height):
+def table_G3_7_lookup(
+    lighting_space_type, occupancy_control_type, space_height, space_area
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/data_fns/table_G3_7_fns.py:123]: I debated about the default values, but I prefer to keep same types/group arguments close to each other. The change only applies to one rule so it is still manageable.

File: rct229/ruleset_functions/normalize_space_schedules.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    assert_required_fields(
+        GET_NORMALIZE_SPACE_SCHEDULE__REQUIRED_FIELDS["space"], space
+    )
+
+    space_total_power_per_area = 0.0
+    space_total_hourly_use_per_area_array = []
+
+    for interior_lighting in space.get("interior_lighting", []):
+        power_per_area = interior_lighting.get(
+            "power_per_area", 0.0
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/ruleset_functions/normalize_space_schedules.py:35]: Make sense. I added an assertion at line 38

File: rct229/ruleset_functions/normalize_space_schedules.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+# Intended for internal use
+GET_NORMALIZE_SPACE_SCHEDULE__REQUIRED_FIELDS = {
+    "space": {
+        "$": ["lighting_space_type", "floor_area"],
+        "interior_lighting[*]": ["lighting_multiplier_schedule"],
+    }
+}
+
+
+def normalize_space_schedules(space, zone_height, schedules):
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/ruleset_functions/normalize_space_schedules.py:14]: Ok, updated

File: rct229/ruleset_functions/normalize_space_schedules.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        schedule_hourly_value = getattr_(
+            find_one_with_field_value(
+                "$", "id", interior_lighting["lighting_multiplier_schedule"], schedules
+            ),
+            "schedule",
+            "hourly_values",
+        )
+        hourly_use_per_area_array = [
+            hourly_value / (1.0 - control_credit) * power_per_area
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/ruleset_functions/normalize_space_schedules.py:61]: Yep, `control_credit` is a fraction of lighting power density so it is always smaller than 1.0 (1.0 means equal to the lighting power density). I will add the comments

File: rct229/ruleset_functions/normalize_space_schedules.py (hunk 1 above)
Comment (weilixu) [reply rct229/ruleset_functions/normalize_space_schedules.py:76]: I think this is black or isort did it...
I am not sure what is more pythonic way to do it..

File: rct229/ruleset_functions/normalize_space_schedules.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                # occupancy control type can be None - simply ignored the credit
+                occupancy_control_type=interior_lighting.get(
+                    "occupancy_control_type", None
+                ),
+                space_height=zone_height,
+                space_area=getattr_(space, "space", "floor_area"),
+            )["control_credit"]
+
+        schedule_hourly_value = getattr_(
+            find_one_with_field_value(
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/ruleset_functions/normalize_space_schedules.py:54]: Good suggestion - I updated the find functions - replace the default assert with the RCT exception assert_ function.

**Comment by weilixu** [comment]
> I think we should eliminate the ONE class from pint_utils.py since there is no single value for one, it depends on the chosen units. It isn't needed here because interior_lighting.power_per_area is already a pint Quantity.
> 
> The ZERO class makes sense because zero is the same for a quantity regardless of the units chosen.

I am getting error if I dont convert the power per area to quantity.
```
self = <Quantity(0.0, 'british_thermal_unit / hour / square_foot')>, other = 2.3
op = <built-in function lt>

    @check_implemented
    def compare(self, other, op):
        if not isinstance(other, Quantity):
            if self.dimensionless:
                return op(
                    self._convert_magnitude_not_inplace(self.UnitsContainer()), other
                )
            elif zero_or_nan(other, True):
                # Handle the special case in which we compare to zero or NaN
                # (or an array of zeros or NaNs)
                if self._is_multiplicative:
                    # compare magnitude
                    return op(self._magnitude, other)
                else:
                    # compare the magnitude after converting the
                    # non-multiplicative quantity to base units
                    if self._REGISTRY.autoconvert_offset_to_baseunit:
                        return op(self.to_base_units()._magnitude, other)
                    else:
                        raise OffsetUnitCalculusError(self._units)
            else:
>               raise ValueError("Cannot compare Quantity and {}".format(type(other)))
E               ValueError: Cannot compare Quantity and <class 'float'>
```

You think it is because of the schema?

**Comment by weilixu** [comment]
> I believe there is only one interior_lighting_schedule, so maybe the function name should be singular.

There could be multiple schedules in one space, each referenced by one interior_lighting data element? However the function will only return one schedule.

**Comment by charliepnnl** [comment]
> > I believe there is only one interior_lighting_schedule, so maybe the function name should be singular.
> 
> There could be multiple schedules in one space, each referenced by one interior_lighting data element? However the function will only return one schedule.

Your call.

File: rct229/ruleset_functions/normalize_interior_lighting_schedules.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    schedules: List[JSON] - schedule element
+    Returns
+    -------
+    A list containing 8760/8784 hourly values of a noralized schedule of the space data element
+    """
+    assert_required_fields(
+        GET_NORMALIZE_SPACE_SCHEDULE__REQUIRED_FIELDS["space"], space
+    )
+
+    space_total_power_per_area = 0.0
```
Comment (charliepnnl) [inline rct229/ruleset_functions/normalize_interior_lighting_schedules.py:32]: This should be `space_total_power_per_area = ZERO.POWER_PER_AREA`.  I'm surprised pint doesn't complain on line 42.

**Pull request merged**